R/tokens_xptr.R

```diff
@@ -83,7 +83,28 @@ types.tokens_xptr <- function(x) {
 
 #' @export
 ntoken.tokens_xptr <- function(x, ...) {
-    structure(cpp_ntoken(x), names = docnames(x))
+    if (length(list(...))) {
+        x2 <- as.tokens_xptr(x)
+        x2 <- tokens(x2, ...)
+    } else {
+        x2 <- x
+    }
+    ret <- structure(cpp_ntoken(x2), names = docnames(x2))
+    rm(x2)
+    ret
+}
+
+#' @export
+ntype.tokens_xptr <- function(x, ...) {
+    if (length(list(...))) {
```

Comment:
It should be like
```
length(types(tokens(x, ...)))
```

Reply:
Not sure what you mean here... the `length()` conditional is just to check whether tokens needs to be called again.

Reply:
`tokens()` does not (should not) do anything if nothing is passed via `...`, so conditioning is redundant.